app/views/shops/index.html.haml

```diff
@@ -2,7 +2,8 @@
   = t :shops_title
 
 - content_for :injection_data do
-  = inject_enterprises(@enterprises)
+  - cache "shops/index/inject_enterprises", expires_in: CacheService::SHOPS_EXPIRY do
+    = inject_enterprises(@enterprises)
```

Comment:
I dont think we should go forward with this without feature tests.
I think we should test both that the cache is working as well as cache invalidation: load the page, wait a second, load again and assert data wasnt reloaded, change the data underneath, reload page and see the fresh data was reloaded.

Reply:
I've just had a look at feature tests for this, but they won't work for any timed caches using the basic `:file_store` or `:memory_store` caching options. We would need to have `memcached` (or something similar) available in the machines the test suite runs on, and on any dev machines where the tests would be run locally.

Reply:
I'm not sure how feasible that would be. It applies to 2 of the 3 PRs, but the "caching by class" one can be covered with feature tests.

Reply:
Actually I'll check the timed caching with basic cache stores, the documentation seems to say conflicting things in different places, maybe it will work...

Reply:
what about a controller test checking a hit and a miss stubbing the underlying store? I don't think it'll be feasible or makes sense getting any deeper than this.

Reply:
When I try to stub the Rails cache I either get errors or it says it hasn't received any calls, even though I can see the cache reads and writes in `logs/test.log`... :disappointed: